enhancements/hypershift/trust-chain-separation.md

```diff
@@ -0,0 +1,203 @@
+---
+title: trust-chain-separation
+authors:
+  - stlaz
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - ibihim
+  - deads2k
+  - alvaroaleman
+  - csrwng
+approvers:
+  - deads2k
+  - alvaroaleman
+  - csrwng
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/AUTH-311
+see-also:
+replaces:
+superseded-by:
+---
+
+# Trust Chain Separation
+
+## Summary
+
+At the time of writing the Hypershift is generating most of its certificates
+by using a single trust-all root CA. This enhancement describes how and why
+such a bundle should be split in order to assure workload portability between
+OpenShift-based products.
+
+## Motivation
+
+The OpenShift certificate trust domain separation has been serving well
+to prevent bugs of extended trust for client certificate authentication,
+and also to prevent implicit trust to servers that the client should not
+attempt connecting to in the first place.
+
+Having a single signer for most certificates is suboptimal as it opens
+window for bugs with unexpected trusts where a server/client grabs the
+global CA bundle and misuses it as this likely "just works" for them.
+It is therefore possible to build a number of solutions which would be
+considered "stable". As that effectively makes the CA bundle an "API",
+it is later hard to split the trust to be more fine-grained as chipping
+away threads of trust breaks the solutions which now rely on the global
+trust bundle.
+
+The intention is to perform at least a minimal split of the client certificate
+signers from the root CA bundle in order to prevent unwanted application
+authentications and to assure portability of user workload among OpenShift
```

Comment:
Thanks for explicitly calling this out.  Portability is the key point for me.  If we miss this opportunity, we will not be able to retrofit without impacting workloads and clients.